src_c/math.c

```diff
@@ -4120,6 +4120,33 @@ math_clamp(PyObject *self, PyObject *const *args, Py_ssize_t nargs)
 
 PG_WRAP_FASTCALL_FUNC(math_clamp, PyObject);
 
+static PyObject *
+math_lerp(PyObject *self, PyObject *const *args, Py_ssize_t nargs)
+{
+    if (nargs != 3)
+        return RAISE(PyExc_ValueError, "lerp requires 3 arguments");
+
+    double a = PyFloat_AsDouble(args[0]);
+    if (PyErr_Occurred())
+        return NULL;
+    double b = PyFloat_AsDouble(args[1]);
+    if (PyErr_Occurred())
+        return NULL;
+    double percent = PyFloat_AsDouble(args[2]);
+    if (PyErr_Occurred())
+        return NULL;
+
+    // Not sure about this. Most other frameworks/engines either automatically
+    // clamp it, or allow you to use values outside this range.
+    // if (percent < 0.0 || percent > 1.0)
+    //    return RAISE(PyExc_ValueError, "Percentage value must be between 0.0
+    //    and 1.0");
```

Comment:
Personally I'm in favour of raising this error, but before you uncomment this, lets wait to see what others have to say

Reply:
Here is some discussion on the topic: https://answers.unity.com/questions/953384/why-is-the-t-value-clamped-to-0-1-in-lerp-function.html

Which sort of makes the point that maybe we should use the clamp function here to clamp the values for percent instead of just erroring if we are going down the "values should only be between 0.0 and 1.0" route. I've personally not intentionally used an unclamped lerp.

Unity seems to provide both options, with the standard lerp being clamped. See:

https://docs.unity3d.com/ScriptReference/Mathf.LerpUnclamped.html

There doesn't seem to be a firm consensus on whether lerp should be clamped or unclamped, I personally don't have a strong preference - I just haven't ever tried to use lerp for extrapolation. Though in the cases I've seen that seems to be mainly used for geometry i.e. lerps on Vectors and maybe colours. What do our current lerps do? Consistency is also a good goal.

Whatever options is chosen (only one function is likely simpler, two is more feature complete - could add another in a separate PR) documentation over what the function does (is it clamped or not) would help.

Reply:
The Vector lerp raises a ValueError if it's outside the range [0, 1]. I guess for now I'll just be consistent and raise it, but I'm still not sure if that's the best idea. It doesn't really line up with most other frameworks.